tempodb/encoding/vparquet4/block_traceql.go

```diff
@@ -3078,9 +3087,14 @@ func (c *batchCollector) KeepGroup(res *parquetquery.IteratorResult) bool {
 				}
 			}
 
-			spans = append(spans, e)
+			if mightFilter {
```

Comment:
i think i'm missing something but is there a reason to do this loop at all if mightFilter/c.requireAtLeastOneMatchOverall is false?

Reply:
I tried using a differently named local var `mightFilter := c.requireAtLeastOneMatchOverall` for readability.  Do you mean to do the rename up one more layer?  Happy to make any readability changes, or maybe even break this out into separate methods, this block still feels a little clunky.